sematic/log_reader.py

```diff
@@ -340,16 +341,33 @@ def load_log_lines(
             reverse=reverse,
             **cursor_kwargs,  # type: ignore
         )
+    else:
+        log_line_result = _load_non_inline_logs(
+            run_id=run_id,
+            still_running=still_running,
+            max_lines=max_lines,
+            filter_strings=filter_strings,
+            default_log_info_message=default_log_info_message,
+            reverse=reverse,
+            **cursor_kwargs,  # type: ignore
+        )
 
-    return _load_non_inline_logs(
-        run_id=run_id,
-        still_running=still_running,
-        max_lines=max_lines,
-        filter_strings=filter_strings,
-        default_log_info_message=default_log_info_message,
-        reverse=reverse,
-        **cursor_kwargs,  # type: ignore
-    )
+    # when no cursor is set, it means that we are doing the initial pull
+    # and we can determine if we can continue in either direction
+    # this overrides the value from the `log_line_result`.
+    if reverse_cursor_token is None and forward_cursor_token is None:
+        if reverse is True:
```

Comment:
```suggestion
        if reverse is True and reverse_cursor_token is None:
```
then you can get rid of the outer `if`. If you're going in reverse from an empty reverse cursor, it doesn't really matter whether the forward cursor is `None` or not--you'll be starting from the very end.

Reply:
what if a user does `reverse=true`, `reverse_cursor_token=none` with `forward_cursor_token=something`? It is an undefined behavior in the newly added logic. So I don't want this code to deal with that scenario, and I want to leave it to the old logic to handle.